tests/unit/testconnectionhealth.cpp

```diff
@@ -78,4 +88,118 @@ void TestConnectionHealth::healthCheckup() {
            ConnectionHealth::ConnectionStability::Stable);
 }
 
-static TestConnectionHealth s_testConnectionHealth;
\ No newline at end of file
+void TestConnectionHealth::testTelemetry() {
+  ConnectionHealth connectionHealth;
+
+  // Nothing recorded at start.
+  metricsTestErrorAndChange(0, 0, 0);
+  metricsTestCount(0, 0, 0);
+  metricsTestTimespan(0, 0, 0);
+
+  // Shouldn't do anything if controller state isn't on.
+  connectionHealth.startActive("", "");
+  connectionHealth.stop();
+  metricsTestErrorAndChange(0, 0, 0);
+  metricsTestCount(0, 0, 0);
+  metricsTestTimespan(0, 0, 0);
+
+  TestHelper::controllerState = Controller::StateOn;
+
+  connectionHealth.setStability(ConnectionHealth::Unstable);
+  connectionHealth.startActive("", "");
+  metricsTestErrorAndChange(0, 1, 0);
+  metricsTestCount(0, 1, 0);
+  metricsTestTimespan(0, 0, 0);
+
+  connectionHealth.setStability(ConnectionHealth::ConnectionStability::Stable);
+  metricsTestErrorAndChange(1, 1, 0);
+  metricsTestCount(1, 1, 0);
+  metricsTestTimespan(0, 1, 0);
+
+  connectionHealth.setStability(
+      ConnectionHealth::ConnectionStability::NoSignal);
+  metricsTestErrorAndChange(1, 1, 1);
+  metricsTestCount(1, 1, 1);
+  metricsTestTimespan(1, 1, 0);
+
+  // stops and sets to stable
+  connectionHealth.stop();
+  metricsTestErrorAndChange(2, 1, 1);
+  metricsTestCount(2, 1, 1);
+  metricsTestTimespan(1, 1, 1);
```

Comment:
Would you mind adding a comment for each of these states? Like you do on line 99? The test becomes a but confusing because of the anonymous arguments for the tester fucntions.

Reply:
Comments added, thanks.

Reply:
Maybe I am missing something but I don't see them?

Reply:
🤦🏻 I forgot to push up the commit. You should see it now.